pkg/controller/agent/fleet.go

```diff
@@ -265,6 +285,34 @@ func maybeReconcileFleetEnrollment(params Params, result *reconciler.Results) En
 	return token
 }
 
+func getKibanaBasePath(ctx context.Context, client k8s.Client, kibanaNSN types.NamespacedName) (string, error) {
+	var kb v1.Kibana
+	err := client.Get(ctx, kibanaNSN, &kb)
+	if err != nil {
+		return "", err
+	}
+
+	if kb.Spec.Config == nil {
+		return "", nil
+	}
+
+	// Get Kibana Config secret
+	kbSecretName := kibana.SecretName(kb)
+	var kbConfigsecret corev1.Secret
+	err = client.Get(ctx, types.NamespacedName{Namespace: kb.Namespace, Name: kbSecretName}, &kbConfigsecret)
+	if err != nil {
+		return "", err
+	}
+
+	kbCfg := kibanaConfig{}
+	err = yaml.Unmarshal(kbConfigsecret.Data[kibana.SettingsFilename], &kbCfg)
+	if err != nil {
+		return "", err
+	}
```

Comment:
This, and the 2 error blocks preceding: I'd suggest we wrap the errors with context of some sort to identify their source/reason.